src/native/sync/omni.rs

```diff
@@ -140,12 +140,13 @@ impl OmniBucket {
             sync_state: SyncState::Unknown,
         };
         // If this bucket already exists both locally and remotely
-        if let Some(bucket) = global.get_bucket(origin)
-            && let Some(remote_id) = bucket.remote_id
-            && RemoteBucket::read(client, remote_id).await.is_ok()
-        {
-            // Prevent the user from re-creating it
-            return Err(NativeError::unique_error());
+        if let Some(bucket) = global.get_bucket(origin) {
+            if let Some(remote_id) = bucket.remote_id {
+                if RemoteBucket::read(client, remote_id).await.is_ok() {
+                    // Prevent the user from re-creating it
+                    return Err(NativeError::unique_error());
+                }
+            }
```

Comment:
Is this invisibly updating the internal global state? Those are scary side-effects that are hard to reason about and violates the expectations of anyone using the library if they're not clearly spelled out. The form of this tells me that I _shouldn't expect side effects_ but the useful values seem to be getting discarded.